example/src/main/java/com/klarna/example/MainActivity.java

```diff
@@ -60,6 +63,7 @@ else if(requestCode == PREFERENCES_REQUEST_CODE) {
     private void updateUIElements() {
```

Comment:
each you do updateUIElements you seek the views which is considered heavy task and the way to avoid it is using members for holding the proper reference. 
i would consider changing to proper members usage.